storage/storage.rs

```diff
@@ -187,10 +188,18 @@ impl<D> MVCCStorage<D> {
         // FIXME proper error
         let mut durability_service = D::recover(path.join(Self::WAL_DIR_NAME)).expect("Could not create WAL directory");
         durability_service.register_record_type::<CommitRecord>();
+        durability_service.register_record_type::<StatusRecord>();
 
         let name = name.as_ref();
         let (keyspaces, keyspaces_index) = recover_keyspaces::<KS>(&storage_dir)?;
-        let isolation_manager = IsolationManager::new(SequenceNumber::from(1));
+
+        let isolation_manager = if durability_service.previous() == SequenceNumber::MIN {
+            let im = IsolationManager::new(Self::tmp_relative_index_from_sequence_number(SequenceNumber::MIN));
+            im.load_aborted(0, SequenceNumber::MIN); // Initialise the watermark to zero
+            im
+        } else {
+            IsolationManager::new(Self::tmp_relative_index_from_sequence_number(durability_service.current()))
```

Comment:
In this branch, we have to recover. In the other branch, we do not right? 
I would move the creation of storage into each `if` and only recover (`.reload()`) when we actually have records.

Reply:
Might be worth discussing, because I do remember us wanting the normal path to be the recovery path.
Maybe it's better to insert an actual committed record when the wal is created.